dashboard/src/components/TimeSelect.tsx

```diff
@@ -0,0 +1,203 @@
+import * as moment from 'moment';
+import * as ReactDatetime from 'react-datetime';
+import CalendarContainer from 'react-datetime/src/CalendarContainer';
+import * as React from 'react';
+import { Button, ButtonGroup, Input, InputGroup,
+         InputGroupButton, Row, Col } from 'reactstrap';
+
+import 'react-datetime/css/react-datetime.css';
+
+import './TimeSelect.css';
+
+export interface Props extends ReactDatetime.DatetimepickerProps {
+    autoUpdate?: boolean;
+    onAutoUpdateChange?: (newValue: boolean) => void;
+}
+
+// The implementation of AutoUpdatingDatetime needs to know more
+// internals of the ReactDatetime than is exported by the TS type
+// definitions.  Make them visible by definining a couple extended
+// interfaces.
+interface ExtendedReactDatetime extends ReactDatetime {
+    getInitialState: () => {};
+    getStateFromProps: (props: Props) => {};
+    onInputKey: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+    onInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    handleClickOutside: () => void;
+}
+interface ReactDatetimeInt {
+    openCalendar?: (event: React.SyntheticEvent<HTMLElement>) => void;
+    onInputChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const ReactDatetimeProto = ReactDatetime.prototype as ExtendedReactDatetime;
+
+class AutoUpdatingDatetime extends ReactDatetime {
+    static defaultProps = {
+        className: '',
+        defaultValue: '',
+        inputProps: {},
+        input: true,
+        onFocus: () => undefined,
+        onBlur: () => undefined,
+        onChange: () => undefined,
+        onViewModeChange: () => undefined,
+        timeFormat: true,
+        timeConstraints: {},
+        dateFormat: true,
+        strictParsing: true,
+        closeOnSelect: false,
+        closeOnTab: true,
+        utc: false,
+
+        onAutoUpdateChange: (newValue: boolean) => undefined,
+        autoUpdate: false,
+    };
+
+    /*constructor(props: Props) {
+        super(props);
+        this.getStateFromProps = this.getStateFromPropsOverride.bind(this);
+    }*/
+
+    getInitialState() {
+        return {
+            ...ReactDatetimeProto.getInitialState.call(this),
+            autoUpdate: AutoUpdatingDatetime.defaultProps.autoUpdate,
+        };
+    }
+
+    getStateFromProps = (props: Props) => {
```

Comment:
Style bikeshedding suggestion:
In the case when using the shorthand notation/new function notation and only returning one value straight away then an even shorter notation can be used:

```
    getStateFromProps = (props: Props) => (
        this.getStateFromPropsOverride(props)
    )
```